Issue with pyinstaller

Can't seem to wrap my head around this.

having a problem compiling a windows exe in python 2.7.9 that uses the requests library and can't find anything on google about the specific error.  My script runs fine from the interpreter but when i use pyinstaller, i get :ImportError: No module named 'requests.packages.chardet.sys'

I can also compile windows executables that don't use requests just fine.

	

    ###Sample Script
    ----------------Begin
    #!/usr/bin/python
     
    import requests
     
    r = requests.get('https://google.com')
     
    print(r.text)
    ----------------End
     
    ###command run to compile into windows exe
    ---------------Begin
    pyinstaller --onefile test.py
    OR
    pyinstaller test.py
    ---------------End
     
    ###Output
    ---------------Begin
    C:\Python27>pyinstaller test.py
    76 INFO: wrote C:\Python27\test.spec
    102 INFO: Testing for ability to set icons, version resources...
    125 INFO: ... resource update available
    129 INFO: UPX is not available.
    164 INFO: Processing hook hook-os
    409 INFO: Processing hook hook-time
    417 INFO: Processing hook hook-cPickle
    536 INFO: Processing hook hook-_sre
    773 INFO: Processing hook hook-cStringIO
    944 INFO: Processing hook hook-encodings
    965 INFO: Processing hook hook-codecs
    1687 INFO: Extending PYTHONPATH with C:\Python27
    1687 INFO: checking Analysis
    1687 INFO: Building Analysis because out00-Analysis.toc non existent
    1688 INFO: running Analysis out00-Analysis.toc
    1690 INFO: Adding Microsoft.VC90.CRT to dependent assemblies of final executable
     
    1781 INFO: Searching for assembly x86_Microsoft.VC90.CRT_1fc8b3b9a1e18e3b_9.0.21
    022.8_none ...
    1782 WARNING: Assembly not found
    1782 ERROR: Assembly x86_Microsoft.VC90.CRT_1fc8b3b9a1e18e3b_9.0.21022.8_none no
    t found
    1954 WARNING: lib not found: MSVCR90.dll dependency of C:\Python27\python.exe
    2039 INFO: Searching for assembly x86_Microsoft.VC90.CRT_1fc8b3b9a1e18e3b_9.0.21
    022.8_none ...
    2040 WARNING: Assembly not found
    2042 ERROR: Assembly x86_Microsoft.VC90.CRT_1fc8b3b9a1e18e3b_9.0.21022.8_none no
    t found
    2263 WARNING: lib not found: MSVCR90.dll dependency of C:\Windows\system32\pytho
    n27.dll
    2266 INFO: Analyzing C:\Python27\lib\site-packages\pyinstaller-2.1.1.dev0-py2.7.
    egg\PyInstaller\loader\_pyi_bootstrap.py
    2284 INFO: Processing hook hook-os
    2309 INFO: Processing hook hook-site
    2339 INFO: Processing hook hook-encodings
    2582 INFO: Processing hook hook-time
    2590 INFO: Processing hook hook-cPickle
    2715 INFO: Processing hook hook-_sre
    2975 INFO: Processing hook hook-cStringIO
    3164 INFO: Processing hook hook-codecs
    3907 INFO: Processing hook hook-pydoc
    4185 INFO: Processing hook hook-email
    4309 INFO: Processing hook hook-httplib
    4368 INFO: Processing hook hook-email.message
    4517 INFO: Analyzing C:\Python27\lib\site-packages\pyinstaller-2.1.1.dev0-py2.7.
    egg\PyInstaller\loader\pyi_importers.py
    4690 INFO: Analyzing C:\Python27\lib\site-packages\pyinstaller-2.1.1.dev0-py2.7.
    egg\PyInstaller\loader\pyi_archive.py
    4865 INFO: Analyzing C:\Python27\lib\site-packages\pyinstaller-2.1.1.dev0-py2.7.
    egg\PyInstaller\loader\pyi_carchive.py
    5040 INFO: Analyzing C:\Python27\lib\site-packages\pyinstaller-2.1.1.dev0-py2.7.
    egg\PyInstaller\loader\pyi_os_path.py
    5069 INFO: Analyzing test.py
    6014 INFO: Processing hook hook-requests
    7263 INFO: Processing hook hook-xml
    7445 INFO: Processing hook hook-xml.sax
    7516 INFO: Processing hook hook-pyexpat
    7646 INFO: Hidden import 'codecs' has been found otherwise
    7648 INFO: Hidden import 'encodings' has been found otherwise
    7648 INFO: Looking for run-time hooks
    7830 WARNING: lib not found: MSVCR90.dll dependency of C:\Python27\lib\site-pack
    ages\win32\win32pipe.pyd
    7987 WARNING: lib not found: MSVCR90.dll dependency of C:\Python27\DLLs\select.p
    yd
    8144 WARNING: lib not found: MSVCR90.dll dependency of C:\Python27\DLLs\unicoded
    ata.pyd
    8319 WARNING: lib not found: MSVCR90.dll dependency of C:\Python27\lib\site-pack
    ages\win32\win32wnet.pyd
    8501 WARNING: lib not found: MSVCR90.dll dependency of C:\Python27\DLLs\_hashlib
    .pyd
    8671 WARNING: lib not found: MSVCR90.dll dependency of C:\Python27\DLLs\bz2.pyd
    8859 WARNING: lib not found: MSVCR90.dll dependency of C:\Python27\DLLs\_ssl.pyd
     
    9052 WARNING: lib not found: MSVCR90.dll dependency of C:\Python27\DLLs\_ctypes.
    pyd
    9223 WARNING: lib not found: MSVCR90.dll dependency of C:\Python27\DLLs\pyexpat.
    pyd
    9460 WARNING: lib not found: MSVCR90.dll dependency of C:\Python27\lib\site-pack
    ages\win32\win32api.pyd
    9632 WARNING: lib not found: MSVCR90.dll dependency of C:\Python27\DLLs\_socket.
    pyd
    9828 WARNING: lib not found: MSVCR90.dll dependency of C:\Windows\system32\pywin
    types27.dll
    9848 INFO: Using Python library C:\Windows\system32\python27.dll
    10016 INFO: Warnings written to C:\Python27\build\test\warntest.txt
    10023 INFO: checking PYZ
    10023 INFO: Rebuilding out00-PYZ.toc because out00-PYZ.pyz is missing
    10024 INFO: Building PYZ (ZlibArchive) out00-PYZ.toc
    12259 INFO: checking PKG
    12261 INFO: Rebuilding out00-PKG.toc because out00-PKG.pkg is missing
    12261 INFO: Building PKG (CArchive) out00-PKG.pkg
    12286 INFO: checking EXE
    12287 INFO: Rebuilding out00-EXE.toc because test.exe missing
    12289 INFO: Building EXE from out00-EXE.toc
    12292 INFO: Appending archive to EXE C:\Python27\build\test\test.exe
    12296 INFO: checking COLLECT
    12296 INFO: Building COLLECT out00-COLLECT.toc
    ---------------End
     
    ###What happens when running the executable
    ---------------Begin

    C:\Users\gRanger\Desktop\dist\test>test.exe
    Traceback (most recent call last):
      File "<string>", line 3, in <module>
      File "c:\python27\lib\site-packages\PyInstaller-2.1.1.dev0-py2.7.egg\PyInstall
    er\loader\pyi_importers.py", line 276, in load_module
        exec(bytecode, module.__dict__)
      File "C:\Users\gRanger\Desktop\build\test\out00-PYZ.pyz\requests", line 58, in
     <module>
      File "c:\python27\lib\site-packages\PyInstaller-2.1.1.dev0-py2.7.egg\PyInstall
    er\loader\pyi_importers.py", line 276, in load_module
        exec(bytecode, module.__dict__)
      File "C:\Users\gRanger\Desktop\build\test\out00-PYZ.pyz\requests.utils", line
    26, in <module>
      File "c:\python27\lib\site-packages\PyInstaller-2.1.1.dev0-py2.7.egg\PyInstall
    er\loader\pyi_importers.py", line 276, in load_module
        exec(bytecode, module.__dict__)
      File "C:\Users\gRanger\Desktop\build\test\out00-PYZ.pyz\requests.compat", line
     7, in <module>
      File "c:\python27\lib\site-packages\PyInstaller-2.1.1.dev0-py2.7.egg\PyInstall
    er\loader\pyi_importers.py", line 276, in load_module
        exec(bytecode, module.__dict__)
      File "C:\Users\gRanger\Desktop\build\test\out00-PYZ.pyz\requests.packages.char
    det", line 19, in <module>
      File "C:\Users\gRanger\Desktop\build\test\out00-PYZ.pyz\requests.packages", li
    ne 83, in load_module
    ImportError: No module named 'requests.packages.chardet.sys'
    ---------------End

I'm afraid I'm still seeing this with requests 2.6.0 and pyinstaller at both version 2.1 and the current master ([67610f](https://github.com/pyinstaller/pyinstaller/commit/67610f2ddadf378c90bf3c8872f3b38baefcb215)). I've created a minimal failing case at [aanand/requests-pyinstaller-bug](https://github.com/aanand/requests-pyinstaller-bug):

```
$ make
script/build-linux
>> Building Docker image
>> Running bin/test
Importing requests
Imported requests 2.6.0
>> Building dist/test
>> Running dist/test
Importing requests
Traceback (most recent call last):
  File "<string>", line 3, in <module>
  File "/usr/local/lib/python2.7/dist-packages/PyInstaller/loader/pyi_importers.py", line 276, in load_module
    exec(bytecode, module.__dict__)
  File "/code/build/test/out00-PYZ.pyz/requests", line 58, in <module>

  File "/usr/local/lib/python2.7/dist-packages/PyInstaller/loader/pyi_importers.py", line 276, in load_module
    exec(bytecode, module.__dict__)
  File "/code/build/test/out00-PYZ.pyz/requests.utils", line 26, in <module>
  File "/usr/local/lib/python2.7/dist-packages/PyInstaller/loader/pyi_importers.py", line 276, in load_module
    exec(bytecode, module.__dict__)
  File "/code/build/test/out00-PYZ.pyz/requests.compat", line 7, in <module>
  File "/usr/local/lib/python2.7/dist-packages/PyInstaller/loader/pyi_importers.py", line 276, in load_module
    exec(bytecode, module.__dict__)
  File "/code/build/test/out00-PYZ.pyz/requests.packages.chardet", line 19, in <module>
  File "/code/build/test/out00-PYZ.pyz/requests.packages", line 95, in load_module
ImportError: No module named 'requests.packages.chardet.sys'
make: *** [default] Error 255
```

--- a/requests/packages/__init__.py
+++ b/requests/packages/__init__.py
@@ -104,4 +104,4 @@ class VendorAlias(object):
         return module
 
 
-sys.meta_path.append(VendorAlias(["urllib3", "chardet"]))
+sys.meta_path.insert(0, VendorAlias(["urllib3", "chardet"]))

